Source/Python/WarpXParticleContainer.cpp

```diff
@@ -29,5 +43,43 @@ void init_WarpXParticleContainer (py::module& m)
             py::arg("nattr_int"), py::arg("attr_int"),
             py::arg("uniqueparticles"), py::arg("id")
         )
-        ;
+        .def("num_real_comps", &WarpXParticleContainer::NumRealComps)
+        .def("total_number_of_particles",
+            &WarpXParticleContainer::TotalNumberOfParticles,
+            py::arg("valid_particles_only"), py::arg("local")
+        )
+        // .def("deposit_charge",
+        //     static_cast<void (WarpXParticleContainer::*)(
+        //         WarpXParIter&,
+        //         amrex::Gpu::DeviceVector<amrex::Real> const &,
+        //         const int * const, amrex::MultiFab*,
+        //         const int, const long, const long,
+        //         const int, const int, const int
+        //     )>(&WarpXParticleContainer::DepositCharge),
+        //     py::arg("pti"), py::arg("wp"), py::arg("ion_lev"),
+        //     py::arg("rho"), py::arg("icomp"),
+        //     py::arg("offset"), py::arg("np_to_depose"),
+        //     py::arg("thread_num"), py::arg("lev"), py::arg("depos_lev")
+        // )
+        .def("deposit_charge",
+            [](WarpXParticleContainer& pc, WarpXParIter& pti,
```

Comment:
Let's try to replace the `&` with `*` for these types to avoid copies.

Reply:
This has been resolved by changing the approach to do the loop in C++ code.